Add development toctree and PyGMT team page (#1308)
GenericMappingTools/pygmt: doc/team.rst

.. title:: Team

Team Gallery
============

We are an international team dedicated to building a Pythonic API for the
Generic Mapping Tools (GMT). Our goal is to improve GMT's accessibility for
new and experienced users by creating user-friendly interfaces with the GMT
C API, supporting rich display in Jupyter notebooks, and integrating with
the PyData ecosystem.

> All our welcome to become involved with the PyGMT project! For more information

Reviewer: Is it a typo?
```suggestion
All are welcome to become involved with the PyGMT project! For more information
```

.. title:: Team

> PyGMT Team Gallery
> ==================

Reviewer: I think "Team Gallery" is enough.
```suggestion
Team Gallery
============
```

We are an international team dedicated to building a Pythonic API for the
Generic Mapping Tools (GMT). Our goal is to improve GMT's accessibility for
new and experienced users by creating user-friendly interfaces with the GMT
C API, supporting rich display in Jupyter notebooks, and integrating with
the PyData ecosystem.

All our welcome to become involved with the PyGMT project! For more information
about how to get involved, see the :doc:`contributing`.

Distinguished Contributors
--------------------------

PyGMT Distinguished Contributors are recognized for their substantial
contributions to PyGMT, which may include code, documentation, pull request
review, triaging, forum responses, community building and engagement,
outreach, and inclusion and diversity. New Distinguished Contributors are

> selected bi-annually by those listed below.

Reviewer: > selected bi-annually by those listed below.

I might miss some discussions, but have you reached an agreement about the frequency? Do you mean "once every two years" or "twice per year"? As a non-native English speaker, it sounds like "once every two years", which is too long for an active project like PyGMT.

When I search for "bi-annually", I found this (https://www.merriam-webster.com/dictionary/biannual):

> When we describe something as biannual, we can mean either that it occurs twice a year or that it occurs once every two years. So how does someone know which particular meaning we have in mind? Well, unless we provide them with a contextual clue, they don't.

Author: Thanks for pointing out the confusing language. I meant twice per year and updated to specify that. It's open for discussion, I just selected 2x per year as an arbitrary starting point.

.. title:: Team

PyGMT Team Gallery
==================

We are an international team dedicated to building a Pythonic API for the
Generic Mapping Tools (GMT). Our goal is to improve GMT's accessibility for
new and experienced users by creating user-friendly interfaces with the GMT
C API, supporting rich display in Jupyter notebooks, and integrating with
the PyData ecosystem.

All our welcome to become involved with the PyGMT project! For more information
about how to get involved, see the :doc:`contributing`.

Distinguished Contributors
--------------------------

> PyGMT Distinguished Contributors are recognized for their substantial

Reviewer 2: See [gmt/AUTHORS.md](https://github.com/GenericMappingTools/gmt/blob/master/AUTHORS.md):
> ## The Founders
> GMT was conceived and designed in the late 1980s and early 1990s by Paul Wessel and Walter H.F. Smith. The > two founders built and maintained GMT from version 1 through 3.

I have another thought. How about adding a short description (or a single category)  for PyGMT founders? That's a great way to show respect to them. I know **Leonardo** is the first to start the PyGMT project, and **Dongdong** and **weiji14** soon joined the project and continue to contribute to the project. We could have more teams and more active contributors in the future, and re-organize the team gallery, but the founders won't change.

Reviewer 3: Leo is mentioned as the founder at https://www.pygmt.org/dev/overview.html. I'm not really a founder, more like an early adopter. We could put a 'Founder' label/tag under Leo's name I suppose.

Author: OK, so then I should probably also add Paul with a 'Founder' tag, since I think him and Leo were both involved with getting the funding to start the project which is a substantial contribution. Sound OK?

Reviewer 2: > OK, so then I should probably also add Paul with a 'Founder' tag, since I think him and Leo were both involved with getting the funding to start the project which is a substantial contribution. Sound OK?

Sound good. But it's better to have more comments/thoughts from @weiji14 and @seisman since they might know more about the history.

Reviewer 3: Most of what I know about PyGMT's early history (or gmt-python as it was back then) is from Leo's Scipy 2017 talk (https://www.youtube.com/watch?v=93M4How7R24). I'm ok with adding Paul in.

Author: Done in https://github.com/GenericMappingTools/pygmt/pull/1308/commits/d0e22aa1370c35cd142910cca1aa3443f8aa7813

.. title:: Team

PyGMT Team Gallery
==================

We are an international team dedicated to building a Pythonic API for the
Generic Mapping Tools (GMT). Our goal is to improve GMT's accessibility for
new and experienced users by creating user-friendly interfaces with the GMT
C API, supporting rich display in Jupyter notebooks, and integrating with
the PyData ecosystem.

All our welcome to become involved with the PyGMT project! For more information
about how to get involved, see the :doc:`contributing`.

> Distinguished Contributors

Reviewer 2: On my mac (13 inch), the display of **Leonardo** is not perfect for 100% view, while 90% is okay. Meanwhile, my workstation is okay since it has a monitor of 23 inch.

100% | 90%
--- | ---
<img width="928" alt="100" src="https://user-images.githubusercontent.com/50591376/122893716-29292d00-d379-11eb-881c-3092875c302a.png"> | <img width="828" alt="90" src="https://user-images.githubusercontent.com/50591376/122893802-3c3bfd00-d379-11eb-9c4d-81b603bb8dce.png">

Author: Thanks for pointing this out. I updated the css style, but unfortunately I cannot figure out how to make the items both a nice grid and also flexible to fit the contents. Please let me know if you think the new style is better.

Reviewer 2: I have the same issue as **Dongdong** pointed in https://github.com/GenericMappingTools/pygmt/pull/1308#issuecomment-867300192 now.

.. title:: Team

PyGMT Team Gallery
==================

> 
> We are an international team dedicating to building a Pythonic API for GMT that makes GMT more accessible to
> new users, interfaces with the GMT C API without system calls, supports rich display in Jupyter notebooks,
> and integrates with the PyData ecosystem.

Reviewer 3: Minor tweaks to the text and some word wrapping (not sure if we should wrap to 79 char or 88 char tbh, I usually do 79 char out of habit).

```suggestion
We are an international team dedicated to building a Pythonic API for the
Generic Mapping Tools (GMT). Our goal is to improve GMT's accessibility for
new users, by creating user-friendly interfaces with the GMT C API, supporting
rich display in Jupyter notebooks, and integrating with the PyData ecosystem.
```

        </div>
    </div>
    </div>
    <div class="team-column">
    <div class="team-card">
        <img class="card-img-top team-img" src="https://avatars.githubusercontent.com/u/23487320?v=4" alt="Card image">
        <div class="team-card-body">
            <h6 class="team-card-text">Wei Ji Leong</h6>
            <p class="team-card-text"><a href="https://github.com/weiji14" >@weiji14</a></p>
        </div>
    </div>
    </div>
    <div class="team-column">
    <div class="team-card">
        <img class="card-img-top team-img" src="https://avatars.githubusercontent.com/u/29518865?v=4" alt="Card image">
        <div class="team-card-body">
            <h6 class="team-card-text";>Will Schlitzer</h6>
            <p class="team-card-text"><a href="https://github.com/willschlitzer" >@willschlitzer</a></p>
        </div>
    </div>
    </div>
    </div>
    <p></p>

Emeritus Maintainers
--------------------

.. raw:: html

    <div class="team-row">
    <div class="team-column">
    <div class="team-card">
        <img class="card-img-top team-img" src="https://avatars.githubusercontent.com/u/290082?v=4" alt="Card image">
        <div class="team-card-body">

>             <h6 class="team-card-text">Leonardo Ueida</h6>

Reviewer 3: Typo on Leo's surname.
```suggestion
            <h6 class="team-card-text">Leonardo Uieda</h6>
```

.. title:: Team

PyGMT Team Gallery
==================

We are an international team dedicated to building a Pythonic API for the
Generic Mapping Tools (GMT). Our goal is to improve GMT's accessibility for
new and experienced users by creating user-friendly interfaces with the GMT
C API, supporting rich display in Jupyter notebooks, and integrating with
the PyData ecosystem.

Maintainers

> -----------

Reviewer 3: 'Maintainers' is correct, but I'd prefer a term like "Core team" or similar so that people won't feel like joining the team means maintaining software only (which is part of our job, but not the whole of it). Any other suggestions?

Reviewer 2: **Core Developer**?

Author: Yes, I agree with your point. I think that specifying who leads project maintenance could be helpful transparency for new contributors, so it would be nice to get a solution that recognizes both maintenance, development, and the rest. Do you think it could be an option to have a 'Distinguished Contributors' section in addition to the 'Maintainers' section, with overlap between the two as appropriate? I like Jupyter's [distinguished contributors](https://jupyter.org/about) section because the path for recognition is clearly described. Or we could do a 'Core Developers' team like @core-man suggested, with some tag for people who are also maintainers. But, it is possible that someone may be interested in being a maintainer without wanting to spend as much time on development.

Author: Just want to follow up on this suggestion @weiji14 and @core-man, do you prefer a single "core team" to having two categories, one for active maintainers and one for core developers/core contributors/distinguished contributors? My preference is the second option, but I am open to either.

Reviewer 2: > I think that specifying who leads project maintenance could be helpful transparency for new contributors, so it would be nice to get a solution that recognizes both maintenance, development, and the rest.

Sounds good. This could help new contributors to have a quick recognition of the PyGMT Team. But the current core developers are all maintainers, right?

> do you prefer a single "core team" to having two categories, one for active maintainers and one for core developers/core contributors/distinguished contributors?

If we only create one team called **core team**, do we have to create a Level-2 title for it (e.g., **Maintainers** and **Emeritus Maintainers** are Level-2 titles now)? Currently, the level-1 title is called **PyGMT Team Gallery**.

We have a few categories and they could be active or not active. I am a little confused about them.
- **Maintainers**
- **Core Developers**
- **Core Contributors**
- **Distinguished Contributors**
- **Contributors**

I think all of you belong to **Core Developers**, and all of you are also maintainers, right? So, I don't think we need to distinguish **Core Developers** and **Maintainers** unless we have one core developer who is not a maintainer, vice versa.

Could we add a labeler, **active**, for those core developers/maintainers who are active now? Thus, it does not need to create a separate category for core developers/maintainers who are not active now.

> My preference is the second option, but I am open to either.

:smile: Sorry that I'm not clear what the first and second options are.

Author: OK, I will make a revision accordingly and then you can check it out to let me know if I understood your comments correctly.

Author: I pushed a revised version based on my interpretation of your comments. I think we need a statement at the top that invites people to get involved and (ideally) lays out the criteria for becoming part of the 'core team' (if this iteration stays). To be honest, while I have spent enough time on PyGMT maintenance to have been comfortable with the maintainers group, it feels odd to me to be the one including myself and listing others on the 'core team' group when I have only been involved with PyGMT for a small amount of time compared to, for example, Wei Ji and Dongdong. I would rather it be better defined somewhere how the 'core team' gets decided and adjusted over time (maybe that guidance exists already, I am not sure).

Reviewer 3: > To be honest, while I have spent enough time on PyGMT maintenance to have been comfortable with the maintainers group, it feels odd to me to be the one including myself and listing others on the 'core team' group when I have only been involved with PyGMT for a small amount of time

It is not the amount of time you've spent, but rather, what you can do that counts :)

At this stage, PyGMT is still a fairly small team (<10 people), and @core-man raises a good point that there's not much of a distinction between developers or maintainers per se. I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.

> Could we add a labeler, active, for those core developers/maintainers who are active now? Thus, it does not need to create a separate category for core developers/maintainers who are not active now. We only have two non-active core developers now.

What about an asterisk (*) after the name to indicate that a person is active? Or have the name be **bold** when someone is active?

Reviewer 4: > At this stage, PyGMT is still a fairly small team (<10 people), and @core-man raises a good point that there's not much of a distinction between developers or maintainers per se. I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.
> 
Fully agree!

> > Could we add a labeler, active, for those core developers/maintainers who are active now? Thus, it does not need to create a separate category for core developers/maintainers who are not active now. We only have two non-active core developers now.
> 
> What about an asterisk (*) after the name to indicate that a person is active? Or have the name be **bold** when someone is active?

If possible, I would prefer somekind of label as suggested by @core-man.

Reviewer 2: I like the current design, and I think any way above to indicate active maintainers/developers are okay. When we have a way to indicate active maintainers/developers soon, new contributors could easily recognize the whole team due to the limited number.

BTW, I feel new contributors could also know the details about PyGMT Teams by [pygmt-team](https://github.com/orgs/GenericMappingTools/teams/pygmt-team/teams), which has a very good and clear layout.

> I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.

It feels odd to me to be included in the PyGMT team gallery list as compared to the big amount of work and time you spent. Meanwhile, I am afraid that I will still have limited time to help with only a few parts of PyGMT (i.e., tutorials/gallery/documentation review, and some module enhancement/wrapping interesting to me (e.g., `plot`/`plot3d`/`sac`/`wiggle`/`histogram` currently))

Reviewer 3: Ok, I see that people are keen on having an **active** label/tag somehow. Will leave it up to @meghanrjones on how to include this in the team gallery and make it look nice :slightly_smiling_face:

> > I'd be tempted to throw @core-man into the PyGMT team gallery list since he's been very helpful with a lot of the documentation and design/decision making.

> It feels odd to me to be included in the PyGMT team gallery list as compared to the big amount of work and time you spent. Meanwhile, I am afraid that I will still have limited time to help with only a few parts of PyGMT (i.e., tutorials/gallery/documentation review, and some module enhancement/wrapping interesting to me (e.g., plot/plot3d/sac/wiggle/histogram currently))

Trust me @core-man, you've probably done more in the past three months in terms of features/documentation/review work than what I did in my first year with PyGMT :laughing:. And it's ok to focus on the parts you use such as the seismological modules. I would encourage you to be on the team list just to show that PyGMT values new contributors that can do good work and isn't restricted to an exclusive club of developers, but I respect your decision if you don't feel ready to be added yet.

Author: https://github.com/GenericMappingTools/pygmt/pull/1308/commits/9dbf864ef42873e83467ddbd47c516b7831f23fb added an 'active' label.

https://github.com/GenericMappingTools/pygmt/pull/1308/commits/52d814123ca516c1aac9094c13534e8972846237 added @core-man. @core-man, I will respect your preferences for whether you want to accept/reject inclusion on the page, so feel free to let me know. I agree with all the previous comments that your contributions are really valuable.

https://github.com/GenericMappingTools/pygmt/pull/1308/commits/d302e67baea0ac5f1d46fe995de932b9d38307dc restructured the page following Jupyter's Distinguished Contributors model, with a label of 'Maintainer' for active maintainers. I prefer this because it meets the two main goals of 1. acknowledging people who have put a lot into the project without requiring any responsibilities from them and 2. providing helpful transparency about who is responsible for the current maintenance of the project. I also prefer having it stated when/how it gets updated. Easy to revert if others do not like it.

Reviewer 2: Thanks a lot for including me in the team gallery. I will continue to do what I can do for the project.

        <div class="team-card-body">
            <h6 class="team-card-text">Michael Grund</h6>
            <p class="team-card-text"><a href="https://github.com/michaelgrund" >@michaelgrund</a></p>
        </div>
    </div>
    </div>
    <div class="team-column">
    <div class="team-card">
        <img class="card-img-top team-img" src="https://avatars.githubusercontent.com/u/23487320?v=4" alt="Card image">
        <div class="team-card-body">
            <h6 class="team-card-text">Wei Ji Leong</h6>
            <p class="team-card-text"><a href="https://github.com/weiji14" >@weiji14</a></p>
        </div>
    </div>
    </div>
    <div class="team-column">
    <div class="team-card">
        <img class="card-img-top team-img" src="https://avatars.githubusercontent.com/u/29518865?v=4" alt="Card image">
        <div class="team-card-body">
            <h6 class="team-card-text";>Will Schlitzer</h6>
            <p class="team-card-text"><a href="https://github.com/willschlitzer" >@willschlitzer</a></p>
        </div>
    </div>
    </div>
    </div>
    <p></p>

Emeritus Maintainers

> --------------------

Reviewer 3: [Emeritus](https://en.wikipedia.org/wiki/Emeritus) sounds a bit old for these two young fellas, is there a better term maybe?

Author: Oops, my bad. Maybe 'Former Maintainers'? Or we could combine the sections with some design choice to indicate activate maintainers.
